### PR TITLE
fix: new colors for multi-connection sidebar COMPASS-7844

### PR DESCRIPTION
**Pull request by paula-stacho**
# === work in progress ===

### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-components/src/components/workspace-tabs/tab.tsx [hunk 1]
```diff
@@ -189,6 +203,14 @@ function Tab({
     props
   );
 
+  const themeClass = useMemo(() => {
+    if (!tabTheme) {
+      return darkMode ? tabDarkThemeStyles : tabLightThemeStyles;
+    }
+
+    return css(tabTheme);
+  }, [tabTheme, darkMode, tabDarkThemeStyles, tabLightThemeStyles, css]);
```
Comment (himanshusinghs) [inline packages/compass-components/src/components/workspace-tabs/tab.tsx:212]:
```suggestion
  }, [tabTheme, darkMode]);
```
Those were outside of the Component so not needed here as deps.

File: packages/compass-connections-navigation/src/placeholder-item.tsx [hunk 2]
```diff
@@ -21,12 +23,18 @@ export const PlaceholderItem: React.FunctionComponent<{
     [level, isSingleConnection]
   );
 
+  const multiConnectionProps = {
+    gradientStart: 'var(--item-bg-color-active)',
+    gradientEnd: 'var(--item-bg-color)',
+    style: { filter: 'brightness(0.98)' },
+  };
```
Comment (himanshusinghs) [inline packages/compass-connections-navigation/src/placeholder-item.tsx:30]: nit: We can probably move it out as its not dependent on anything in the Component?

File: packages/compass-connections-navigation/src/utils.ts [hunk 3]
```diff
@@ -28,10 +28,10 @@ export const getItemPaddingStyles = ({
     /** MC version */
     switch (level) {
       case 2:
-        paddingLeft = spacing[3];
+        paddingLeft = spacing[3] + spacing[1] + spacing[50];
```
Comment (himanshusinghs) [inline packages/compass-connections-navigation/src/utils.ts:31]: This made me realise that I need to read on those new tokens for spacing.

File: packages/connection-form/src/hooks/use-connection-color.ts [hunk 4]
```diff
@@ -83,15 +126,38 @@ export function legacyColorsToColorCode(
...
   connectionColorCodes: () => ColorCode[];
   connectionColorToHex: (colorCode: string | undefined) => string | undefined;
+  connectionColorToHexActive: (
+    colorCode: string | undefined
+  ) => string | undefined;
   connectionColorToName: (colorCode: string | undefined) => string | undefined;
 } {
   const newColorCodeToHex = useCallback(
     (colorCode: string | undefined): string | undefined => {
+      const isDarkMode = useDarkMode();
```
Comment (himanshusinghs) [inline packages/connection-form/src/hooks/use-connection-color.ts:136]: We probably should move this hook outside of this function and have it just once for both this useCallback and the one below it.

**Review by himanshusinghs — APPROVED** [review]
Leaving a few suggestions but it looks good. 🚀 
Colors yayy 🥳

File: packages/compass-connections-navigation/src/utils.ts (hunk 3 above)
Comment (himanshusinghs) [reply packages/compass-connections-navigation/src/utils.ts:31]: nothing to do here

File: packages/compass-components/src/components/resizeable-sidebar.tsx [hunk 5]
```diff
@@ -26,10 +26,10 @@ const containerStylesDark = css({
   '--icon-color': palette.white,
 
   '--item-color': palette.white,
-  '--item-color-active': palette.green.base,
+  '--item-color-active': palette.white,
```
[the remark this answers is not in the document]
Comment (himanshusinghs) [reply packages/compass-components/src/components/resizeable-sidebar.tsx]: I think this has gone now - did not see this anywhere.

File: packages/compass-sidebar/src/components/legacy/sidebar.tsx [hunk 6]
```diff
@@ -175,11 +176,11 @@ export function Sidebar({
   return (
     <ResizableSidebar
       data-testid="navigation-sidebar"
-      className={sidebarStyles}
+      className={cx(sidebarStyles)}
```
Comment (himanshusinghs) [inline packages/compass-sidebar/src/components/legacy/sidebar.tsx:179]:
```suggestion
      className={sidebarStyles}
```

File: packages/compass-connections-navigation/src/tree-item.tsx [hunk 7]
```diff
@@ -113,6 +113,12 @@ export function useDefaultAction<T>(
 const itemContainer = css({
   cursor: 'pointer',
   color: 'var(--item-color)',
+  backgroundColor: 'var(--item-bg-color)',
+  backgroundRadious: 'var(--item-bg-radius)',
```
Comment (himanshusinghs) [inline packages/compass-connections-navigation/src/tree-item.tsx:117]:
```suggestion
  backgroundRadius: 'var(--item-bg-radius)',
```

File: packages/compass-sidebar/src/components/multiple-connections/saved-connections/saved-connection.tsx [hunk 8]
```diff
@@ -223,6 +223,7 @@ export function SavedConnection({
       }}
       className={savedConnectionStyles}
       data-testid={`saved-connection-${connectionInfo.id}`}
+      onDoubleClick={() => onConnect(connectionInfo)}
```
Comment (paula-stacho) [inline packages/compass-sidebar/src/components/multiple-connections/saved-connections/saved-connection.tsx:226]: ❤️

**Comment by paula-stacho** [comment]
@kmruiz we also want to remove the bar on the left of active elements, could you do it here? if it turns out to be tricky we can of course leave it for the follow up with the other tough parts  😅 

Screenshot 2024-04-30 at 10 19 35

**Comment by paula-stacho** [comment]
what happened to moving the colors out of connection-form? do we have a follow up for it?

File: packages/compass-connections-navigation/src/styled-navigation-item.tsx [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  isSingleConnection,
+  children,
+}: {
+  colorCode?: string;
+  isSingleConnection: boolean;
+  children: React.ReactChild;
+}): React.ReactElement {
+  const { connectionColorToHex, connectionColorToHexActive } =
+    useConnectionColor();
+  const style: React.CSSProperties & {
```
Comment (paula-stacho) [inline packages/compass-connections-navigation/src/styled-navigation-item.tsx:18]: should this be a memo? sorry, I know I wrote this bit, but now that I'm having a second look on it.. 😁

Comment (kmruiz) [reply packages/compass-connections-navigation/src/styled-navigation-item.tsx:18]: haha no worries, let me take a look at it :D

**Review by paula-stacho — COMMENTED** [review]
I can't press the approve button cause I opened the request originally 😆 

✅ ✅

**Pull request merged**